src/data-processor.ts

```diff
@@ -0,0 +1,45 @@
+const R         = require('ramda');
+
+/**
+ * Parses users objects and outputs a formatted string.
+ * Called by other functions when wanting to output the result of a query
+ * @param data User objects to parse
+ */
+function processDataToFormat(data: any[]){
+    let processedUsers = {};
+    let outputFormat = R.map((user) => {
+                            if(!R.has(user.login, processedUsers)){
+                                processedUsers[user.login] = true;
+                                return R.reduce(R.concat,'',['Login: ',user.login,
+                                ' || Name: ',user.name,
+                                ' || Company: ',(user.company || 'Undisclosed'),
+                                ' || Liked Languages: ', languageByUser(user.login,data)]);
+                            }
+                            else
+                                return "";
+                        });                            
+          
+    let outputArray = R.filter((line) => line != "",outputFormat(data));
+  
+    R.map((out) => console.log("##",out))(outputArray);
+}
```

Comment:
To increase this function's readability you could declare the `map` function **outside** the function and just call it inside of it.
Something like:
```
const mapFn = val => console.log(val);
.
.
.

let outputFormat = R.map(mapFn);
```
It'll become much more readable, and less like something close to a [callback hell staircase](http://callbackhell.com/)!